controller/RestEligibility.php

```diff
@@ -322,32 +355,41 @@ public function get()
      * Get eligibility instance from request
      * @return Eligibility
      * @throws \common_exception_MissingParameter
-     * @throws \common_exception_NotFound
+     * @throws common_exception_RestApi
      */
     private function getEligibilityFromRequest()
     {
-        $eligibility = $this->getParameterFromRequest(self::PARAMETER_ELIGIBILITY_ID);
-        $resource = $this->getAndCheckResource($eligibility, EligibilityService::CLASS_URI);
-        $eligibility = $this->propagate(new Eligibility($resource->getUri()));
-        return $eligibility;
+        try {
+            $eligibilityUri = $this->getParameterFromRequest(self::PARAMETER_ELIGIBILITY_ID);
+            $resource = $this->getAndCheckResource($eligibilityUri, EligibilityService::CLASS_URI);
+            $eligibility = $this->propagate(new Eligibility($resource->getUri()));
+
+            return $eligibility;
+        } catch (common_exception_NotFound $e) {
+            throw new common_exception_RestApi(__('Eligibility `%s` does not exist.', $eligibilityUri), 404);
+        }
     }
 
     /**
      * Get delivery resource from request parameters
      * @return \core_kernel_classes_Resource
      * @throws \common_exception_MissingParameter
-     * @throws \common_exception_NotFound
+     * @throws common_exception_RestApi
      */
     private function getDeliveryFromRequest()
     {
-        $deliveryUri = $this->getParameterFromRequest(self::PARAMETER_DELIVERY_ID);
-        return $this->getAndCheckResource($deliveryUri, DeliveryAssemblyService::CLASS_URI);
+        try {
+            $deliveryUri = $this->getParameterFromRequest(self::PARAMETER_DELIVERY_ID);
+
+            return $this->getAndCheckResource($deliveryUri, DeliveryAssemblyService::CLASS_URI);
+        } catch (common_exception_NotFound $e) {
+            throw new common_exception_RestApi(__('Delivery `%s` does not exist.', $deliveryUri));
```

Comment:
To keep consistency, you could return `404` here as well, just like in case of `getEligibilityFromRequest()`.